mogp_emulator/demos/excalibur_workshop_demo.py

```diff
@@ -0,0 +1,133 @@
+
+gp = mogp_emulator.GaussianProcess(simulation_points, simulation_output)
+gp = mogp_emulator.fit_GP_MAP(gp)
+
+print("Zero mean and no priors:")
+print("Correlation lengths = {}".format(np.sqrt(np.exp(-gp.theta[:2]))))
+print("Sigma = {}".format(np.sqrt(np.exp(gp.theta[2]))))
+print("Nugget = {}".format(gp.nugget))
+print()
+
+# We can look at how the emulator performs by comparing the emulator output to
+# a large number of validation points. Since this simulation is cheap, we can
+# actually compute this for a large number of points.
+
+n_valid = 1000
+validation_points = lhd.sample(n_valid)
+validation_output = np.array([simulator(p) for p in validation_points])
+
+if makeplots:
+    import matplotlib.tri
+    tri = matplotlib.tri.Triangulation((validation_points[:,0]+4.)/4.,
+                                       (validation_points[:,1]/1000.))
+
+    plot_solution(validation_output, "True simulator", "simulator_output.png",
+                  simulation_points, validation_points, tri)
+
+# Now predict values with the emulator and plot output and error
+
+predictions = gp.predict(validation_points)
+
+if makeplots:
+    plot_solution(predictions.mean, "MLE emulator", "emulator_output_MLE.png",
+                  simulation_points, validation_points, tri)
+
+# This is not very good! The simulation points are too sparsely sampled to give the
+# emulator any idea what to do about the function shape. We just know the value at a few
+# points, and it throws up its hands and predicts zero everywhere else.
+
+# To improve this, we will specify a mean function and some priors to ensure that if we are
+# far away from an evaluation point we will still get some information from the emulator.
+
+# We specify the mean function using a string, which follows a similar approach to R-style
+# formulas. There is an implicit constant term, and we use x[index] to specify how we
+# want the formula to depend on the inputs. We choose a simple linear form here, which has
+# three fitting parameters in addition to the correlations lengths, sigma, and nugget
+# parameters above.
+
+meanfunc = "x[0]+x[1]"
+
+# We now set priors for all of the hyperparameters to better constrain the estimation procedure.
+# We assume normal priors for the mean function parameters with a large variance (to not constrain
+# our choice too much). Note that the mean function parameters are on a linear scale, while the
+# correlation lengths, sigma, and nugget are on a logarithmic scale. Thus, if we choose normal
+# priors on the correlation lengths, these will actually be lognormal distributions.
+
+# Finally, we choose inverse gamma and gamma distributions for the priors on sigma and the nugget
+# as those are typical conjugate priors for variances/precisions. We pick them to be where they are as
+# we expect sigma to be large (as the function is very sensitive to inputs) while we want the
+# nugget to be small.
```

Comment:
This explanation feels a little incomplete - perhaps it could be either elaborated on or removed (and the (Inv)Gamma priors used without comment).  The second sentence is helpful, so maybe this paragraph could begin "We choose distributions for the sigma and the nugget parameters with the property that..."

Reply:
I agree -- I originally tried to write out a longer explanation but it ended up being too convoluted and confusing so I scrapped it and swept it under the rug because I was never happy with it.

The larger problem is that the raw parameter values are rather opaque, and each are scaled a bit differently to ensure we have a more robust optimization problem. I haven't had a chance to think carefully about how to best manage it (see issue #89 for some scattered thoughts), but I feel tackling that is going to be important for better user comprehension and usability. Then, we need to think about the implications for priors and how to parametrize those, and this quickly becomes a trickier problem than I anticipated.

So anyways, just noting that you would be correct in your comment, but at the moment I don't have a good way for the code itself to make this more transparent, which I think should be a goal.